### ISSUE TITLE
svelte/no-navigation-without-resolve does not allow `ResolvedPathname` as href

### ISSUE THREAD

**Issue by AndersRobstad** [sveltejs/eslint-plugin-svelte#1319]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.34.0

### What version of `eslint-plugin-svelte` are you using?

3.12.1

### What did you do?

<details>

No real config for this rule used, its included in the svelte recommended rules

</details>

```svelte
<script lang="ts">
	import { type ResolvedPathname } from '$app/types';

	interface Props {
		href: ResolvedPathname;
	}

	const { href }: Props = $props();
</script>

<!-- This shows error `Found a link with a url that isn't resolved.` -->
<a href={href}>
     Link here to a resolved pathname
</a>

```

### What did you expect to happen?

I would expect to be able to pass in href's into a link which has types that imply that they are resolved, and thus valid hrefs for the application. This is very useful when a components link's are determined by where its being used, where its really handy to be able to just pass it a resolved href from the layout that is rendering it. 

### What actually happened?

I'm not able to do this which makes this rule very tedious to use

### Link to **GitHub Repo** with Minimal Reproducible Example

Repro not needed as its shown in simple code above

**Comment by marekdedic** [comment]
Thanks, that seems like a false positive, will take a closer look :)

**Comment by polyzen** [comment]
Also perhaps worth noting, this rule does not seem to trigger if you use `<a {href}>foo</>`.

Edit: does **not***

**Comment by marekdedic** [comment]
@polyzen aah, you think you thought of all the edge cases and then after it makes it to recommended, there's so many more.... :D

In other words, thanks, I'll take a look.

**Comment by cdcarson** [comment]
This should be an opt-in rule. It's a breaking change for ci tasks that require linting to pass.

**Comment by marekdedic** [comment]
Every change to the recommended rule set is a breaking change for CI tasks and as far as I understand, the policy of this package is that that is okay in minor versions.

**Comment by cdcarson** [comment]
> Every change to the recommended rule set is a breaking change for CI tasks

True. Every change is **potentially** breaking to somebody's project.  But not every change is created equal:

1. Scope. This change affects _every SvelteKit project_ with `<a href="/foo">Foo</a>` links. It therefore has a very large blast zone.
2. Utility. Those sites were presumably working just fine without this rule. Projects that were concerned with type-safe URLs could already use the type checking available through `resolve('/foo')`. The rule has limited utility.

**Comment by marekdedic** [comment]
Thanks for the response. Ultimately, this is for the package maintainers to decide (and I'm not one of them :)), but my 2 cents:

1. Scope: Great, most rules in this package affect *every Svelte project*, so this is quite limited in comparison :D
2. Utility: IMO, this rule fixes a massive foot-gun in SvelteKit. In fact, the SvelteKit team is considering just doing this in SvelteKit by default (sveltejs/kit#11803), but it's a breaking change, so I assume they are waiting for v3.

**Comment by AndersRobstad** [comment]
> This should be an opt-in rule. It's a breaking change for ci tasks that require linting to pass.

We often just make new rules emit warnings instead of errors if we don't want to deal with the errors of a new rule right away. This means we still get the benefits the new rule brings instantly, but we can deal with the warnings when we have the time for it. You still get reminded about the warnings every time you open a file with the warning, or when you inspect the lint output.

**Comment by axel7083** [comment]
We are also facing this problem, and the only solution seems to use `<!-- eslint-disable svelte/no-navigation-without-resolve -->` which is a bit sad to be to add everywhere
